Docs: add access and authorize log fields configuration settings (#906)
pomerium/documentation: content/docs/reference/access-log-fields.mdx

![Add custom headers in the Console](./img/http-logs/access-custom-headers.gif)

</TabItem>
<TabItem label="Kubernetes" value="Kubernetes">

| **Name** | **Type** | **Default** |
| :-- | :-- | :-- |
| `accessLogFields` | Array of strings | See [Access Fields and Defaults](#access-fields-and-defaults) |

</TabItem>
</Tabs>

## Access Log Fields and Defaults

The table below lists all available access log fields:

| **Field name** | **Description** | **Default field** |
| :-- | :-- | :-- |
| `authority` | The HTTP Authority. Can be a hostname, domain name, or IP address and may contain a port number (for example, `www.example.com`) | Yes |
| `duration` | The amount of time the request takes to complete in seconds | Yes |
| `forwarded-for` | Identifies the originating IP address of the client | Yes |
| `method` | The HTTP method, such as `GET`, `POST`, or `PUT` | Yes |
| `path` | The HTTP path (for example, `/some/path`) | Yes |
| `referer` | The HTTP Referer, or the address of the web page from which the resource has been requested | Yes |
| `request-id` | The request's unique identifier | Yes |
| `response-code` | The HTTP Response Code (for example `200` or `500`) | Yes |

> | `response-code-details` | The HTTP Response Code Details (for example, `File not found` or `404`) | Yes |

Reviewer: I think this is actually set by Envoy, and unrelated to the HTTP response itself. It looks like the documentation for this field is here: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/response_code_details.

> | `size` | The HTTP Response size | Yes |

Reviewer: Maybe add "in bytes" here?

> | `upstream-cluster` | The cluster the request was sent to | Yes |

Reviewer: We might want to mention here that the cluster name is assigned by Pomerium and/or give some examples of what this might look like (e.g. "route-3bb74f76f9d71f9c" or "pomerium-control-plane-http").

| `user-agent` | The User-Agent field sent by the user's browser | Yes |
| `query` | The HTTP query (for example, `?test=one&other=13`) | No |
| \* [`headers.{CustomHeaderName}`](#custom-headers-fields) | An HTTP Request Header identified by the `HeaderName` (for example, `headers.X-Amzn-Trace-Id` might return `Root=1-64c03960-37809588421513e42f260f56`) | No |

\* The `headers.{CustomHeaderName}` field is only available in **Core** and **Enterprise**, not Kubernetes.

## Access log behavior

### Find access logs

> To find access logs, in your terminal search for `"message":"http-request"`:

Reviewer: I think production deployments may not necessarily be running in a terminal, so I might suggest phrasing this like "To find access logs, filter your Pomerium logs for entries containing `"message": "http-request"`."

Also, I believe access log entries are logged by the `proxy` service, so we might want to mention that here. Maybe something like this? "In split service mode, access log entries are logged by the Proxy service."

```json
{
  "level": "info",
  "service": "envoy",
  "upstream-cluster": "route-6f431f0be965b608",
  "method": "GET",
  "authority": "verify.pomerium.com",
  "path": "/img/json.svg",
  "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36",
  "referer": "https://verify.localhost.pomerium.io/index.css",
  "forwarded-for": "XXX.XXX.X.XXX",
  "request-id": "6f3ad640-cb15-48b4-b79f-9adb94dd654c",
  "duration": 224.860959,
  "size": 5127,
  "response-code": 200,
  "response-code-details": "via_upstream",
  "time": "2023-08-03T16:55:14-04:00",
  "message": "http-request"
}
```

### Display specific logs

> If `access_log_fields` is not defined, all the available default fields will be displayed.

Reviewer: nit: "displayed" -> "logged"

| **Field name** | **Description** | **Default field** |
| :-- | :-- | :-- |
| `authority` | The HTTP request `:authority` or `Host` header value. Can be a domain name or IP address and may contain a port number (for example, `www.example.com`) | Yes |
| `duration` | The amount of time the request takes to complete in seconds | Yes |

> | `forwarded-for` | Identifies the originating IP address of the client (this field is the value of the [`X-Forwarded-For`](/docs/reference/x-forwarded-for-http-header) header as sent by the upstream service) | Yes |

Reviewer: ```suggestion
| `forwarded-for` | Identifies the originating IP address of the client (this field is the value of the [`X-Forwarded-For`](/docs/reference/x-forwarded-for-http-header) header as sent to the upstream service) | Yes |
```

| `method` | The HTTP request method, such as `GET`, `POST`, or `PUT` | Yes |
| `path` | The HTTP request path (for example, `/some/path`) | Yes |
| `referer` | The HTTP request referer, or the address of the web page from which the resource has been requested | Yes |
| `request-id` | The request's unique identifier as assigned by Pomerium | Yes |
| `response-code` | The HTTP Response Code (for example `200` or `500`) | Yes |

> | `response-code-details` | The HTTP Response Code Details set by Envoy (for example, `File not found` or `404`; see [Envoy docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/response_code_details) for more information) | Yes |

Reviewer: ```suggestion
| `response-code-details` | The response code details set by Envoy (for example, `via_upstream` or `route_not_found`; see the [Envoy docs](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/response_code_details) for more information) | Yes |
```

### Display specific logs

If `access_log_fields` is not defined, all the available default fields will be logged.

> To display only a specific field, or multiple fields, add the desired field(s) to your configuration file.

Reviewer: ```suggestion
To log only a specific field, or multiple fields, add the desired field(s) to your configuration file.
```
